### ISSUE TITLE
How to understand ngram_range=(3, 3)?

### ISSUE THREAD

**Issue by ahbon123** [znwang25/fuzzychinese#7]
May I ask how to understand `ngram_range=(3, 3)`? What if I change its values into `(1, 2)`?

**Comment by znwang25** [comment]
This is inherited from `sklearn`. See [here](https://scikit-learn.org/stable/modules/feature_extraction.html).

The idea is to breakdown a string into pieces, which we would call ["n-grams"](https://en.wikipedia.org/wiki/N-gram). 
An n-gram of size 1 is referred to as a "unigram"; size 2 is a "bi-gram". 

For example, if we want to breakdown the sentence, `"Bi-grams are cool!"`,  into just bi-grams (`ngram_range=(2, 2)`), we would get 
`['bi grams', 'grams are', 'are cool']`. 

If we set the `ngram_range=(1, 2)`, we would get both unigrams and bi-grams.
`['bi', 'grams', 'are', 'cool', 'bi grams', 'grams are', 'are cool']`

In this package, we are trying to breakdown a string of Chinese characters into ngrams. If you choose the analyzer to be `"char"`, then each unigram will be a character. If you choose `"stroke"`, then each unigram will be a stroke. 

Choosing a smaller `ngram_range` will break the string into more pieces and may slow down the calculation. But it may (or may not) improve the match accuracy. It's a balancing act.
